internal/config/resource_base_sdkv2.go

```diff
@@ -0,0 +1,198 @@
+package config
+
+import (
+	"context"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func NewAnalyticsResourceSDKv2(d *schema.Resource, name string) *schema.Resource {
+	analyticsResource := &AnalyticsResourceSDKv2{
+		resource: d,
+		name:     name,
+	}
+	/*
+		We are not initializing deprecated fields, for example Update to avoid the message:
+			resource mongodbatlas_cloud_backup_snapshot: All fields are ForceNew or Computed w/out Optional, Update is superfluous
+
+		Ensure no deprecated fields are used by running `staticcheck ./internal/service/... | grep -v 'd.GetOkExists'` and looking for (SA1019)
```

Comment:
Would this be something to take in considerations when upgrading terraform-plugin-sdk?